test/tools/yulopti.cpp

```diff
@@ -139,98 +145,39 @@ class YulOpti
 				m_nameDispenser = make_shared<NameDispenser>(m_dialect, *m_ast, reservedIdentifiers);
 				disambiguated = true;
 			}
-			cout << "(q)uit/(f)latten/(c)se/initialize var(d)ecls/(x)plit/(j)oin/(g)rouper/(h)oister/" << endl;
-			cout << "  (e)xpr inline/(i)nline/(s)implify/varname c(l)eaner/(u)nusedprune/ss(a) transform/" << endl;
-			cout << "  (r)edundant assign elim./re(m)aterializer/f(o)r-loop-init-rewriter/for-loop-condition-(I)nto-body/" << endl;
-			cout << "  for-loop-condition-(O)ut-of-body/s(t)ructural simplifier/equi(v)alent function combiner/ssa re(V)erser/" << endl;
-			cout << "  co(n)trol flow simplifier/stack com(p)ressor/(D)ead code eliminator/(L)oad resolver/" << endl;
-			cout << "  (C)onditional simplifier/conditional (U)nsimplifier/loop-invariant code (M)otion?" << endl;
+			map<char, string> const& abbreviationMap = OptimiserSuite::stepAbbreviationToNameMap();
```

Comment:
Yeah, you have a point. A function working with scalars would hide implementation details better and would still be perfectly usable for me. I'll change it.

Reply:
I changed the functions to:

``` c++
static char stepNameToAbbreviation(std::string const& stepName);
static std::string const& stepAbbreviationToName(char stepAbbreviation);
```
but now I'm starting to see some disadvantages of this solution:
- With a `map` I could easily check if a character has a corresponding step. Now I can't unless I change the return value to something like `string const*` (not great), `optional<string>` (can't use a reference here) or add another function just for checking if the abbreviation is present in the map.
- I had to keep one of the functions creating the map (private). Otherwise I would have to either duplicate the name-abbreviation map or have one function awkwardly reconstruct it by calling the other in a loop.
- Being able to pass this map to the function printing the list of available steps in `yulopti` was nice. Now I'll have to do it differently.

Given the above, I think the solution with two maps was simpler and I would prefer to keep it. What do you think?